dataset/src/main/scala/frameless/TypedColumn.scala

```diff
@@ -471,6 +471,31 @@ abstract class AbstractTypedColumn[T, U]
   def cast[A: TypedEncoder](implicit c: CatalystCast[U, A]): ThisType[T, A] =
     typed(self.untyped.cast(TypedEncoder[A].catalystRepr))
 
+  /**
+    * An expression that returns a substring
+    * {{{
+    *   df.select(df('a).substr(0, 5))
```

Comment:
Question: in Spark, is it possible to we mix literals and columns? (`.substr(0, df('a))`)

Reply:
nope, there are only `substr(sartPos: Int, len: Int)` and `substr(startPos: Column, len: Column)`